Improvement: not clearing the name after modal with search results was canceled

Hurhaj opened the pull request:
This PR closes #180.

- when the results of a search is  saved, input from user is then cleared
- when its canceled, input from user is not cleared, user can continue adjusting his input

Agreon commented on client/components/AddGameModal.tsx:
```diff
@@ -228,7 +228,10 @@ const SetupGameForm: React.FC<{ onClose: () => void }> = ({ onClose }) => {
                     isLoading={loading}
                     disabled={loading || game.syncing}
-                    onClick={() => onAdd({ name, continueSearching })}
+                    onClick={() => {
+                        onAdd({ name, continueSearching });
+                        onSave();
+                    }}
```
A better approach would be to replace the `onSuccess: onClose` in line 132 with `onSucess: onSave` so that we only call this method if the `onAdd` really worked. With the current implementation here we do not wait for the promise of `onAdd` to resolve

Agreon reviewed (changes requested):
Nice work! Thanks a lot. Please see my one comment and fix the `eslint` issues. You can install the `eslint` plugin, that may help.

Hurhaj commented:
Hi again! I changed the logic of the save button. I also installed eslint as you suggested, and fixed the issues. But the workflow did not run yet, could you start the work flow so that i can see if my eslint changes were resolved?

Agreon commented:
It seems to be okay now :) Thank you very much!
